common/src/main/java/com/tc/net/core/BufferManager.java

```diff
@@ -27,12 +27,40 @@
  * @author Ludovic Orban
  */
 public interface BufferManager {
+  /**
+   * Called immediately after a call to recvToBuffer(). It will be called again if it returns a non-zero value.
+   * It is called from the connection's read thread.
+   * This method should write data into the supplied ByteBuffer.
+   * @param dest the ByteBuffer into which data should be written
+   * @return the number of bytes written
+   */
   int forwardFromReadBuffer(ByteBuffer dest);
 
+  /**
+   * Called when there is data to write to the channel and the channel is accepting writes.
+   * It is called from the connection's write thread.
+   * This method should read data from the supplied ByteBuffer.
+   * @param src the ByteBuffer from which data should be read
+   * @return the number of bytes read
+   */
   int forwardToWriteBuffer(ByteBuffer src);
 
+  /**
+   * Called immediately after a call to a forwardToWriteBuffer() method. It will be called repeatedly until the cumulative
+   * total of bytes sent matches the number of bytes returned by the forwardToWriteBuffer() call.
```

Comment:
`It will be called repeatedly until the cumulative total of bytes sent matches the number of bytes returned by the forwardToWriteBuffer() call.` is again an implementation detail of the comm stack.